plumbing/format/config/encoder.go

```diff
@@ -59,7 +59,7 @@ func (e *Encoder) encodeSubsection(sectionName string, s *Subsection) error {
 func (e *Encoder) encodeOptions(opts Options) error {
 	for _, o := range opts {
 		pattern := "\t%s = %s\n"
-		if strings.Contains(o.Value, "\\") {
+		if strings.Contains(o.Value, "\\") || strings.Contains(o.Value, "#") {
```

Comment:
There are more characters that could cause problems with config values.
Also see #354 for an approved PR for this issue.